electron/gog/games.ts

```diff
@@ -0,0 +1,368 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+import { GOGLibrary } from './library'
+import { BrowserWindow } from 'electron'
+import Store from 'electron-store'
+import { spawn } from 'child_process'
+import { join } from 'path'
+import prettyBytes from 'pretty-bytes'
+import { Game } from '../games'
+import { GameConfig } from '../game_config'
+import { GlobalConfig } from '../config'
+import {
+  ExtraInfo,
+  GameInfo,
+  InstallInfo,
+  GameSettings,
+  ExecResult,
+  InstallArgs,
+  LaunchResult,
+  GOGLoginData,
+  InstalledInfo
+} from 'types'
+import { existsSync, rmSync } from 'graceful-fs'
+import {
+  gogdlBin,
+  heroicGamesConfigPath,
+  isWindows,
+  execOptions
+} from '../constants'
+import { logError, logInfo, LogPrefix } from '../logger/logger'
+import { errorHandler, execAsync } from '../utils'
+import { GOGUser } from './user'
+import { launch } from '../launcher'
+import { addShortcuts, removeShortcuts } from '../shortcuts'
+
+const configStore = new Store({
+  cwd: 'gog_store'
+})
+
+const installedGamesStore = new Store({
+  cwd: 'gog_store',
+  name: 'installed'
+})
+
+class GOGGame extends Game {
+  public appName: string
+  public window = BrowserWindow.getAllWindows()[0]
+  private static instances = new Map<string, GOGGame>()
+  private constructor(appName: string) {
+    super()
+    this.appName = appName
+  }
+  public static get(appName: string) {
+    if (!this.instances.get(appName)) {
+      this.instances.set(appName, new GOGGame(appName))
+    }
+    return this.instances.get(appName)
+  }
+  public async getExtraInfo(namespace: string): Promise<ExtraInfo> {
+    const gameInfo = GOGLibrary.get().getGameInfo(this.appName)
+    const extra: ExtraInfo = {
+      about: gameInfo.extra.about,
+      reqs: await GOGLibrary.get().createReqsArray(
+        this.appName,
+        gameInfo.is_mac_native && process.platform == 'darwin'
```

Comment:
This is unreadable. Would be better to write a function to return that or use another thing like a switch.